### DISCUSSION TITLE
Setter returned by useAtom causes rerender

### DISCUSSION THREAD

**Discussion by m-irc-o** [pmndrs/jotai#1549] [Q&A]
When using an atom's setter inside a `useEffect` with adding it to the dependency array causes the effect to run on every render.  
Not adding the setter to the dependency array solves the problem, but there is the nasty suggestion of eslint to add it to the dependencies.

Is there an elegant way to get around this?

**Comment by dai-shi** [comment]
Can you create a small reproduction with codesandbox https://react.new ?

**Reply by m-irc-o** [reply]
Well after updating from version 1.7.6 to 1.9.1 this isn't a problem anymore.
Thx for the quick reply
